rmw_zenoh_cpp/src/detail/zenoh_utils.hpp

```diff
@@ -18,31 +18,28 @@
 #include <zenoh.h>
 
 #include <chrono>
-#include <functional>
 #include <optional>
 
 #include "rmw/types.h"
 
 namespace rmw_zenoh_cpp
 {
 ///=============================================================================
-// A function to safely copy an entity's GID as a z_bytes_t into a
-// z_owned_bytes_map_t for a given key.
-using GIDCopier = std::function<void (z_owned_bytes_map_t *, const char *)>;
-///=============================================================================
-z_owned_bytes_map_t
-create_map_and_set_sequence_num(int64_t sequence_number, GIDCopier gid_copier);
+void
+create_map_and_set_sequence_num(
+  z_owned_bytes_t * out_bytes, int64_t sequence_number,
+  uint8_t gid[RMW_GID_STORAGE_SIZE]);
```

Comment:
The `GIDCopier` helps to ensure the entitie's GID is being locked first before we access it to copy. Why do we change that behavior here? as far as I can tell, we jut need to replace `z_owned_bytes_map_t *` with `z_owned_bytes_t *`?

Reply:
Yes. That was my thought as well. But sorry about the sudden change without highlighting it before.